csharp-advanced-topics/FuncActionDelegate/FuncActionDelegate/Program.cs

```diff
@@ -0,0 +1,58 @@
+﻿public class Program
+{
+    public class Candidate
+    {
+        public string Name { get; set; }
+        public int Age { get; set; }
+        public int Experience { get; set; }
+    }
+
+    private static bool CheckExperience(Candidate candidate)
+    {
+        return candidate.Experience >= 3; // 5 scores
+    }
+
+    private static bool CheckAge(Candidate candidate)
+    {
+        return candidate.Age < 40; // 5 scores
+    }
+
+    private static void DisplayScore(Candidate candidate, int score)
+    {
+        Console.WriteLine($"{candidate.Name}'s score is {score}");
+    }
+
+    private static void EvaluateCandidates(List<Candidate> candidates, List<Func<Candidate, bool>> checks, Action<Candidate, int> displayScore)
+    {
+        foreach (var candidate in candidates)
+        {
+            int score = 0;
+
+            foreach (var check in checks)
+            {
+                if (check(candidate))
+                    score += 5;
+            }
+
+            displayScore(candidate, score);
+        }
+    }
```

Comment:
Either in a separate class or in the Candidate class. You shouldn't put so much logic in the Program class.